validator/sawtooth_validator/state/batch_tracker.py

```diff
@@ -93,6 +125,28 @@ def get_statuses(self, batch_ids):
         with self._lock:
             return {b: self.get_status(b) for b in batch_ids}
 
+    def get_invalid_txn_info(self, batch_id):
```

Comment:
Nothing uses any invalid txn functionality in this commit. It just adds the functionality. This method is the only way to access that new functionality. It would only make sense to move it to the next commit (where it is used), if the two commits were combined.